diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4262ce16..a6cf70cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -196,7 +196,7 @@ target_include_directories(tjs PUBLIC src)
 add_executable(tjs-cli
     src/cli.c
 )
-target_link_libraries(tjs-cli tjs)
+target_link_libraries(tjs-cli tjs -static)
 set_target_properties(tjs-cli
     PROPERTIES OUTPUT_NAME tjs
 )
@@ -357,6 +357,7 @@ set(WAMR_BUILD_LIB_PTHREAD 0)
 set(WAMR_BUILD_MINI_LOADER 0)
 set(WAMR_DISABLE_HW_BOUND_CHECK 1)
 set(WAMR_DISABLE_STACK_HW_BOUND_CHECK 1)
+set(WAMR_BUILD_TARGET ARM)
 
 # Include WAMR build scripts
 include(${WAMR_ROOT_DIR}/build-scripts/runtime_lib.cmake)
